### PR TITLE
#57 Add search by resource Year of Release

### PR DESCRIPTION
**Pull request by abulojoshua1**
#### what does the PR do
- Adds Search by Year of Release

#### Changes Made
- Removed implementation for year of custom year of release in dataset page
- Indexed custom year of release in dataset resource

### CODE REVIEW

File: ckanext/faoclh/plugin.py [hunk 1]
```diff
@@ -22,17 +22,17 @@
...
 
 FIELD_RESOURCE_TYPE = 'fao_resource_type'
 FIELD_ACTIVITY_TYPE = 'fao_activity_type'
 FIELD_FOCUS         = 'fao_geographic_focus'
 
-FIELD_RELEASE_YEAR  = 'fao_release_year'
 FIELD_RESOURCE_YEAR = 'fao_resource_release_year'
 
-VOCAB_FIELDS = [FIELD_RESOURCE_TYPE, FIELD_ACTIVITY_TYPE, FIELD_FOCUS]
+VOCAB_FIELDS = [FIELD_RESOURCE_TYPE, FIELD_ACTIVITY_TYPE, FIELD_FOCUS, FIELD_RESOURCE_YEAR]
```
Comment (etj) [inline ckanext/faoclh/plugin.py:35]: Please double check if the FIELD_RESOURCE_YEAR can really be added here, in a list related to VOCABULARIES.
We also expect the items in this list to be fields of the dataset, and FIELD_RESOURCE_YEAR isn't.

Comment (abulojoshua1) [reply ckanext/faoclh/plugin.py:35]: @etj, I did the above cause I could not find a better way to add the `fao_resource_release_year` in facet list on side nav in the dataset search page and display all the years of release that can be used to filter through datasets.

Comment (etj) [reply ckanext/faoclh/plugin.py:36]: @abulojoshua1 Facets list is handled in `_fao_facets()` method

Note that this list is used in the `_modify_package_schema` method, so the FIELD_RESOURCE_YEAR is added as a dataset field, and it should not.

Comment (abulojoshua1) [reply ckanext/faoclh/plugin.py:36]: I'll make a change to the `_fao_facets()` method instead.

**Pull request merged**